WordPress/src/main/java/org/wordpress/android/ui/stories/StoryMediaSaveUploadBridge.kt

```diff
@@ -0,0 +1,170 @@
+        // let's invoke the UploadService and enqueue all the files that were saved by the FrameSaveService
+        val frames = StoryRepository.getStoryAtIndex(saveResult.storyIndex).frames
+        val uriList = frames.map { Uri.fromFile(it.composedFrameFile) }
+        addNewMediaItemsToPostAsync(site, uriList)
+    }
+
+    private fun addNewMediaItemsToPostAsync(site: SiteModel, uriList: List<Uri>) {
+        // this is similar to addNewMediaItemsToEditorAsync in EditorMedia
+        launch {
+            addLocalMediaToPostUseCase.addNewMediaToEditorAsync(
+                    uriList,
+                    site,
+                    freshlyTaken = false, // we don't care about this
+                    editorMediaListener = this@StoryMediaSaveUploadBridge,
+                    doUploadAfterAdding = true
+            )
+            postUtils.preparePostForPublish(editPostRepository.getEditablePost()!!, site)
+            savePostToDbUseCase.savePostToDb(WordPress.getContext(), editPostRepository, site)
+
+            if (networkUtils.isNetworkAvailable()) {
+                postUtils.trackSavePostAnalytics(
+                        editPostRepository.getPost(),
+                        site
+                )
+                uploadService.uploadPost(WordPress.getContext(), editPostRepository.id, true)
+                // SAVED_ONLINE
+            } else {
+                // SAVED_LOCALLY
+            }
+        }
+    }
+    // endregion
+
+    private fun cancelAddMediaToEditorActions() {
+        job.cancel()
+    }
+
+    @Subscribe(sticky = true, threadMode = MAIN)
+    fun onEventMainThread(event: StorySaveResult) {
+        // only trigger the bridge preparation and the UploadService if the Story is now complete
+        // otherwise we can be receiving successful retry events for individual frames we shoulnd't care about just
+        // yet.
+        if (isStorySuccessfullySavedAndComplete(event)) {
+            // only remove it if it was successful - we want to keep it and show a snackbar once when the user
+            // comes back to the app if it wasn't, see MySiteFrament for details.
+            EventBus.getDefault().removeStickyEvent(event)
+            event.metadata?.let {
+                val site = it.getSerializable(WordPress.SITE) as SiteModel
+                editPostRepository.loadPostByLocalPostId(it.getInt(StoryComposerActivity.KEY_POST_LOCAL_ID))
+                addNewStoryFrameMediaItemsToPostAndUploadAsync(site, event)
+                // TODO WPSTORIES add TRACKS
+                // lets add an EVENT for START UPLOADING MEDIA
+                // AnalyticsTracker.track(Stat.STORIES_BLA_BLA_ADDED_MEDIA_OR_SOMETHING);
+            }
+        } else {
+            // TODO WPSTORIES add TRACKS for ERROR
+            // AnalyticsTracker.track(Stat.MY_SITE_ICON_UPLOAD_UNSUCCESSFUL);
+        }
+    }
+
+    private fun isStorySuccessfullySavedAndComplete(event: StorySaveResult): Boolean {
```

Comment:
Nitpick : Wondering if this name can be simplified 🙏 
```suggestion
    private fun isStorySavingComplete(event: StorySaveResult): Boolean {
```

Reply:
👍 addressed in 414beab72f